### ISSUE TITLE
Question: How to pass html content?

### ISSUE THREAD

**Issue by pratiksyngenta** [alonrbar/easy-template-x#54]

**Comment by alonrbar** [comment]
Hi, 
Not sure I understand exactly what you are asking but generally speaking the library doesn't support HTML input. If you can specify more precisely what you mean, what is the input and what is the expected output I can try and give a more accurate response.

**Comment by rigoramos** [comment]
@thegnuu I'd like to know if you could embed the HTML in the word

**Comment by thegnuu** [comment]
I could yes, technically it is quite easy to render HTML content.

Unfortunately it is not so easy to that in a generic way. The main problem here is the way Word is handling its styles. There are basically two possible ways to go:

1. Use word styles
2. Apply custom styles depending on the HTML element directly to rendered content

Version 1 is kinda hard to achieve since every element needs to be mapped to a proper style - unfortunately those styles depends on each file which means you have to add custom styles and you need to make sure those are embedded and language agnostic. So the only way to use this method is to use the same word template for all of your docs. And you need to specify which elements should be mapped to which style.

Version 2 works good for simple elements like `<h1>`, `<p>` and to apply properties like colors, unfortunately it works not well on elements like eg. bullet lists, since those elements require styles like in version 1.

Long story short: It possible to render HTML content but for now I have not found a way to simply achieve this goal in a generic way that would allow me to write a plugin that everyone can use.

**Comment by alonrbar** [comment]
You might want to check the new plugin by @cvaize. The plugin currently implements `<br>`, `<b>`, `<i>` and `<u>` and can be found here: https://github.com/cvaize/easy-template-x-extension-html.
